website/src/_posts/2021-12-2.1-2.3.md

```diff
@@ -0,0 +1,100 @@
+
+<video alt="Audio plugin demo" muted autoplay loop>
+  <source src="/images/blog/2.1-2.3/unsplash-demo.mp4" type="video/mp4">
+  Your browser does not support the video tag: https://uppy.io/images/blog/2.1-2.3/unsplash-demo.mp4
+</video>
+
+PR [#3196](https://github.com/transloadit/uppy/pull/3196).
+
+## Refactor locale scripts & generate types and docs
+
+This is an internal change, but important nonetheless. Our locales now live in a separate `locale.js` file for every plugin instead of `this.defaultLocale` in the class. The docs and types are automatically updated when we add/remove new locale strings.
+
+See PR [#3276](https://github.com/transloadit/uppy/pull/3276) for details.
+
+## Status bar error state improvements
+
+Status bar plugin would get confused about upload errors in Uppy, and we’ve addressed that:
+
+* Error state is shown if only _one_ failed file instead of _all_ failed files. Used to be stuck in the uploading state when a file failed.
+* Added “x of x files uploaded” below “Upload failed” for extra context.
+* Improved the error details button styling in the statusbar and the file info card.
+* Status bar state is set to complete if the user manually removes the failed files.
+
+![status bar improvements screenshot](/images/blog/2.1-2.3/status-bar-improvements.jpg)
+
+PR [#3299](https://github.com/transloadit/uppy/pull/3299).
+
+## Yarn v3 instead of npm
+
+We’ve switched the uppy repo to Yarn 3 to improve package install performance — with our complex dependency graph and over 30 packages in a monorepo, we are seeing a speed increase of over a few minutes!
+
+We are using Corepack to make sure all Uppy contributors are using the same version of Yarn. If you are an Uppy contributor, please run the following commands to enable Corepack:
+
+```sh
+corepack -v || npm i -g corepack # install Corepack if not already installed
+yarn -v || corepack enable # install Yarn through Corepack if not already installed
+yarn || corepack yarn install # install npm dependencies using yarn
+```
+
+See details in [#3237](https://github.com/transloadit/uppy/pull/3237)
+
+## Misc
+
+* @uppy/google-drive: plugin has been updated for brand compliance (#3178)
+* @uppy/status-bar: now shows all details on mobile when `showProgressDetails` is `true` (#3174)
+* @uppy/core: move `Uppy` class to its own module (#3225)
+* @uppy/angular: fix component crash by loosening `package.json` version constraints (#3210)
+* @uppy/drop-target: Exposed events (#3238)
+
+## Companion
+
+### Simultaneous upload/download #3159
+
+This removes the need for disk space as data will be buffered in memory and backpressure will be respected. This is especially useful for large files (multi-GB).
+
+**NOTE:** This is turned off by default (`streamingUpload` / `COMPANION_STREAMING_UPLOAD` option). Do not turn on yet if using TUS for uploads, due to [this tus-js-client bug](https://github.com/tus/tus-js-client/issues/275)
```

Comment:
This should get a lot more love in my opinion. It's a good feature and was quite a bit of work! Move it up and explain in better detail why this is good. Perhaps even some benchmarks if its noticeable and we really want to impress people.

cc @mifi @kvz

Reply:
Thanks! I gave it some more love now, and merged the "more" section from companion with the other one.